.semaphore/semaphore.yml

```diff
@@ -26,10 +26,30 @@ blocks:
     dependencies: []
     run:
       # don't run the tests on non-functional changes...
-      when: "change_in('/', {exclude: ['/.deployed-versions/', '.github/'], default_branch: 'master'})"
+      when: "change_in('/', {exclude: ['/.deployed-versions/', '.github/'], default_branch: 'main'})"
     task:
       jobs:
         - name: Test & Build
           commands:
             - go test ./...
             - make build
+
+  - name: "Build and deploy docker image"
+    dependencies: []
+    run:
+      when: "change_in(['Dockerfile'], {pipeline_file: 'ignore'})"
+    task:
+      jobs:
+        - name: "Build and deploy docker image"
+          commands:
+            - set -x
+            - if [ "$SEMAPHORE_GIT_WORKING_BRANCH" = "main" ]; then IMAGE_ENV=prod; else IMAGE_ENV=dev; fi
+            - export IMAGE_REPO=519856050701.dkr.ecr.us-west-2.amazonaws.com/docker/${IMAGE_ENV}/confluentinc/boring-registry
+            - export IMAGE="${IMAGE_REPO}:${SEMAPHORE_GIT_SHA}"
+            - BUILDKIT_PROGRESS=plain docker build -t $IMAGE .
+            - |
+              if [ "$SEMAPHORE_GIT_WORKING_BRANCH" = "main" ]; then
+                docker tag $IMAGE $IMAGE_REPO:latest
+                docker push $IMAGE_REPO:latest
+              fi
```

Comment:
We shouldn't need to tag `latest` in this use case. We should always tag using a semantic version because we will always deploy using a semantic version.
```suggestion
```
 I guess we need to sort out versioning as well... https://github.com/confluentinc/confluent-bazel-rules/blob/master/python/scripts/bump_version.sh is the typical way we handle this. I am not sure what a good light weight way to do this is. Maybe a fine hacky approach right now is to clone https://github.com/confluentinc/confluent-bazel-rules and run python/scripts/bump_version.sh

Reply:
Actually, the python one wouldn't work because we don't have the python files. It looks like https://github.com/confluentinc/confluent-bazel-rules/blob/master/.semaphore/bump_version.sh would work

Reply:
since this is a go repo, should we use cc-mk-include to build and publish with versioning?

Reply:
It looks like the upstream uses goreleaser to build binaries with a versioning logic. I'll see if I can add docker build with goreleaser and reuse the logic

Reply:
We can't use cc-mk-include for public repos